test/ibmq/test_ibmq_job_attributes.py

```diff
@@ -239,21 +271,21 @@ def _mocked__api_job_submit(*args, **kwargs):
     def test_queue_info(self, provider):
         """Test retrieving queue information."""
         # Find the most busy backend.
+        # States a job could be in when queued.
+        queue_states = [JobStatus.QUEUED, JobStatus.RUNNING]
         backend = most_busy_backend(provider)
         qobj = assemble(transpile(self._qc, backend=backend), backend=backend)
-        leave_states = list(JOB_FINAL_STATES) + [JobStatus.RUNNING]
         job = backend.run(qobj, validate_qobj=True)
         queue_info = None
         for _ in range(10):
             queue_info = job.queue_info()
-            # Even if job status is QUEUED, queue information may not be
-            # immediately available.
-            if (job._status is JobStatus.QUEUED and job.queue_position() is not None) or \
-                    job._status in leave_states:
+            # Even if job status is queued, its queue info may not be immediately available.
+            if (job._status in queue_states and job.queue_position() is not None) or \
+                    job._status in JOB_FINAL_STATES:
```

Comment:
I don't understand the logic here. The old code says "if job is QUEUED and has queue position, or if the job progressed beyond QUEUED state, then break". The new one says "if job is QUEUED and RUNNING and has queue position"? But the job should not be in RUNNING if there's a queue position?

Reply:
I mistook it for the case where the JobStatus may display as RUNNING, but the job is actually QUEUED. I'll change it back to the original logic.